### PR TITLE
another attempt to fix ToXContent

### PR DESCRIPTION
**Pull request by mkhludnev**
I'm trying to guess logic behind XContent fragments, but I rather think it's worth to check with someone who better understand it.

### CODE REVIEW

**Comment by mkhludnev** [comment]
@macohen @epugh ping

File: src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureSet.java [hunk 1]
```diff
@@ -44,7 +45,7 @@
 import static org.apache.lucene.util.RamUsageEstimator.NUM_BYTES_OBJECT_HEADER;
 import static org.apache.lucene.util.RamUsageEstimator.NUM_BYTES_OBJECT_REF;
 
-public class StoredFeatureSet implements FeatureSet, Accountable, StorableElement {
+public class StoredFeatureSet implements FeatureSet, Accountable, StorableElement, ToXContentObject {
     public static final int MAX_FEATURES = 10000;
```
Comment (macohen) [inline src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureSet.java]: StorableElement extends ToXContent; maybe it should also implement ToXContentObject?

File: src/main/java/com/o19s/es/ltr/logging/LoggingSearchExtBuilder.java [hunk 2]
```diff
@@ -77,9 +77,11 @@ public static LoggingSearchExtBuilder parse(XContentParser parser) throws IOExce
...
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
-        builder.field(LOG_SPECS.getPreferredName(), logSpecs);
-        return builder.endObject();
+        //builder.startObject(); //since it's marked as XContentFragment,
+        // so Strings.toString(org.opensearch.core.xcontent.MediaType, org.opensearch.core.xcontent.ToXContent, org.opensearch.core.xcontent.ToXContent.Params, boolean, boolean)
+        // embraces fragment
+        return  builder.field(LOG_SPECS.getPreferredName(), logSpecs);
+        //return builder.endObject();
     }
```
Comment (macohen) [inline src/main/java/com/o19s/es/ltr/logging/LoggingSearchExtBuilder.java:85]: not sure if this is always a fragment. I also don't think I trust the test suite to tell us. line coverage is at 45% in com.o19.es.

**Comment by macohen** [comment]
I'd love to get some more eyes on this. @noCharger? @msfroh mentioned he could take a look...

**Comment by macohen** [comment]
Oh; and thanks so much for checking this out @mkhludnev!

File: src/main/java/com/o19s/es/ltr/logging/LoggingSearchExtBuilder.java (hunk 2 above)
Comment (mkhludnev) [reply src/main/java/com/o19s/es/ltr/logging/LoggingSearchExtBuilder.java:85]: `logSpecs` is a list, so it's hardly might be a fragment 
`LoggingSearchExtBuilder.LogSpec implements ToXContentObject` and has ` builder.startObject();` in `toXContent()` so. they are not fragments.

File: src/test/java/com/o19s/es/ltr/feature/store/StoredFeatureParserTests.java [hunk 3]
```diff
@@ -98,7 +98,8 @@ public void testToXContent() throws IOException {
         String featureString = generateTestFeature();
         StoredFeature feature = parse(featureString);
         XContentBuilder builder = MediaTypeRegistry.contentBuilder(XContentType.JSON);
-        featureString = String.valueOf(feature.toXContent(builder, ToXContent.EMPTY_PARAMS));
+        //featureString = String.valueOf(feature.toXContent(builder, ToXContent.EMPTY_PARAMS));
+        featureString = Strings.toString(XContentType.JSON, feature);//.toXContent(builder, ToXContent.EMPTY_PARAMS));
```
Comment (mkhludnev) [inline src/test/java/com/o19s/es/ltr/feature/store/StoredFeatureParserTests.java:102]: @macohen align test with `StoredFeatureSetParserTests.testToContent()`

File: src/test/java/com/o19s/es/ltr/feature/store/StoredFeatureSetParserTests.java [hunk 4]
```diff
@@ -88,8 +88,6 @@ public void testToXContent() throws IOException {
         String featureSetString = generateRandomFeatureSet("my_set", features::add);
         StoredFeatureSet featureSet = parse(featureSetString);
 
-        XContentBuilder builder = XContentFactory.jsonBuilder();//contentBuilder(XContentType.JSON);
```
Comment (mkhludnev) [inline src/test/java/com/o19s/es/ltr/feature/store/StoredFeatureSetParserTests.java:91]: no use

File: src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureSet.java (hunk 1 above)
Comment (mkhludnev) [reply src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureSet.java]: just did that in the last commit, also had to tweak tests

**Pull request merged**